src/llm/default.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import asyncio
+
+from llm.abstract_llm import AbstractLLM
+from learn.idf import categorize_conversation_history
+from llm.memory import generate_memory_instance
+from llm.embeddings import get_most_relevant_contents_from_message
+
+from langchain.chains.llm import LLMChain
+from langchain.memory.chat_memory import BaseChatMemory
+
+from langchain.chat_models import ChatOpenAI
+from langchain.memory import ConversationBufferWindowMemory
+from langchain.prompts import (ChatPromptTemplate, HumanMessagePromptTemplate,
+                               MessagesPlaceholder,
+                               SystemMessagePromptTemplate)
+
+from settings import OPENAI_API_KEY
+
+
+class DialogLLM(AbstractLLM):
+    @property
+    def memory(self) -> BaseChatMemory:
+        if self.session_id:
+            return generate_memory_instance(
+                session_id=self.session_id,
+                parent_session_id=self.config.get("parent_session_id")
```

Comment:
I think that `parent_session_id` should be injected as a new parameter on instance initialization